### PR TITLE
Yaml config

### PR DESCRIPTION
**Pull request by alexmaul**
Enable scheduler to read a yaml-formatted configuration file.
The old config-format is still supported.

### CODE REVIEW

File: trollsched/schedule.py [hunk 1]
```diff
@@ -42,12 +42,8 @@
 
 logger = logging.getLogger(__name__)
 
-# shortest allowed pass in minutes
-MIN_PASS = 4
-
 # name/id for centre/org creating schedules
-# CENTER_ID = "SMHI"
-CENTER_ID = "DWD-OF"
+CENTER_ID = "SMHI"
```
Comment (mraspaud) [inline trollsched/schedule.py]: I thought we used schedule.CENTER_ID ?

File: scheduler.yaml_template [hunk 2]
```diff
@@ -0,0 +1,127 @@
+default:
+    center: SMHI
```
Comment (mraspaud) [inline scheduler.yaml_template]: in the cfg file I call it `center_id`, so we should make it consistent by either changing in the cfg or in th yaml

File: trollsched/utils.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        sat_scores = {}
+        for sat in satellites:
+            sat_scores[sat] = (cfg["satellites"][sat]["night"],
+                                             cfg["satellites"][sat]["day"])
+
+        station_list.append(((station_lon, station_lat, station_alt),
+                station_name, area, sat_scores))
+
+    foo = cfg["default"].get("center")
+    if foo:
```
Comment (mraspaud) [inline trollsched/utils.py]: `foo` is usually a test variable, can we make it more explicit ?

File: scheduler.yaml_template (hunk 2 above)
Comment (mraspaud) [reply scheduler.yaml_template]: ...or that's the way I wrote it in the documentation. So it has to be changed somewhere.

File: trollsched/utils.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    try:
+        return read_config_yaml(filename)
+    except yaml.parser.ParserError as e:
+        return read_config_cfg(filename)
+
+def read_config_cfg(filename):
+    """Read the config file *filename* and replace the values in global
+    variables.
+    """
+    station_list = []
```
Comment (mraspaud) [inline trollsched/utils.py]: `center` should be read here.

File: trollsched/utils.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        for sat in satellites:
+            sat_scores[sat] = (cfg["satellites"][sat]["night"],
+                                             cfg["satellites"][sat]["day"])
+
+        station_list.append(((station_lon, station_lat, station_alt),
+                station_name, area, sat_scores))
+
+    foo = cfg["default"].get("center")
+    if foo:
+        schedule.CENTER_ID = foo
```
Comment (mraspaud) [inline trollsched/utils.py]: This doesn't seem to be used in the `generate_schedule_xml` function

File: scheduler.yaml_template (hunk 2 above)
Comment (alexmaul) [reply scheduler.yaml_template]: ok, center_id is better.

File: trollsched/schedule.py (hunk 1 above)
Comment (alexmaul) [reply trollsched/schedule.py]: I hadn't looked at your commit to develop, this line is obsolete then.

File: trollsched/utils.py (hunk 4 above)
Comment (alexmaul) [reply trollsched/utils.py]: yes.

File: trollsched/utils.py (hunk 5 above)
Comment (alexmaul) [reply trollsched/utils.py]: It doesn't indeed. I've used the sort-of static schedule.CENTER_ID to stay compatible with the cfg-file, where the center_id wasn't set.
With your commit in develop these lines might not be necessary?

Comment (mraspaud) [reply trollsched/utils.py]: Well, I pass center_id as an argument, I think your approach is superior.

**Pull request merged**